openedx/features/enterprise_support/signals.py

```diff
@@ -0,0 +1,33 @@
+"""
+This module contains signals related to enterprise.
+"""
+
+from django.dispatch import receiver
+from django.db.models.signals import post_save
+from django.contrib.auth.models import User
+
+from enterprise.models import EnterpriseCustomerUser
+from email_marketing.models import EmailMarketingConfiguration
+from email_marketing.tasks import update_user
+from email_marketing.signals import create_sailthru_user_vars, get_current_site
+
+
+@receiver(post_save, sender=EnterpriseCustomerUser)
+def email_marketing_enterprise_user(sender, instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Called after enterprise learner is created and saved
+
+    Args:
+        sender: Not used
+        instance: The enterprise user object
+        kwargs: Not used
+    """
+    user = User.objects.get(id=instance.user_id)
+
+    email_config = EmailMarketingConfiguration.current()
+    if not email_config.enabled:
+        return
```

Comment:
You don't need this block here. It is already done when you call the `update_user` task.